src/secureSignal_shared.ts

```diff
@@ -0,0 +1,108 @@
+const MAXIMUM_RETRY = 3;
+export class UidSecureSignalProvider {
+  debug: boolean;
+
+  constructor(debug = false) {
+    this.debug = debug;
+
+    if (typeof window.getUidAdvertisingToken === 'function') {
+      this.logging('register SecureSignalProvider');
+      this.registerSecureSignalProvider();
+    }
+  }
+
+  public registerSecureSignalProvider = () => {
+    const uidHandler = this.retrieveAdvertisingTokenHandler();
+
+    if (!uidHandler) {
+      console.warn('UidSecureSignal: Please implement `getUidAdvertisingToken`');
+      return;
+    }
+
+    window.googletag = window.googletag || {
+      cmd: [],
+    };
+
+    window.googletag.secureSignalProviders = window.googletag.secureSignalProviders || [];
+    window.googletag.secureSignalProviders.push({
+      id: 'uidapi.com',
```

Comment:
We'll need this to be euid.eu for the EUID version.

Reply:
changed